services/ux-backend/handlers/onboarding_tokens.go

```diff
@@ -0,0 +1,110 @@
+package handler
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/red-hat-storage/ocs-operator/v4/services/types"
+)
+
+func OnboardingTokensHandler(w http.ResponseWriter, r *http.Request) {
+
+	switch r.Method {
+	case "POST":
+		onboardingToken, err := getOnboardingToken()
+		if err != nil {
+			http.Error(w, encodeResponse(types.HTTPResponse{Code: http.StatusMethodNotAllowed, Message: onboardingToken}), http.StatusInternalServerError)
+			return
+		}
+
+		onboarding := struct {
+			Token string `json:"token"`
+		}{}
+
+		onboarding.Token = onboardingToken
+		marshaledOnboardingToken, err := json.Marshal(onboarding)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		//nolint
+		w.Write(marshaledOnboardingToken)
+
+	default:
+		http.Error(w, encodeResponse(types.HTTPResponse{Code: http.StatusMethodNotAllowed, Message: "Only POST request is allowed"}), http.StatusMethodNotAllowed)
+		return
+	}
+}
+
+func getOnboardingToken() (string, error) {
+
+	tokenLifetimeInHours, err := strconv.Atoi(os.Getenv("ONBOARDING_TOKEN_LIFETIME"))
+	if err != nil {
+		tokenLifetimeInHours = 48
+	}
+
+	expirationDate := time.Now().Add(time.Duration(tokenLifetimeInHours) * time.Hour).Unix()
+
+	payload, err := json.Marshal(types.OnboardingTicket{
+		ID:             uuid.New().String(),
+		ExpirationDate: expirationDate,
+	})
+
+	if err != nil {
+		return "Failed to marshall payoad", err
+	}
+
+	encodedPayload := base64.StdEncoding.EncodeToString(payload)
+	// Before signing, we need to hash our message
+	// The hash is what we actually sign
+	msgHash := sha256.New()
+	_, err = msgHash.Write(payload)
+	if err != nil {
+		return "Failed to hash payload", err
+	}
+
+	msgHashSum := msgHash.Sum(nil)
+	pemString, err := os.ReadFile("/etc/private-key/key")
+	if err != nil {
+		return "Failed to read private key", err
+	}
+
+	// In order to generate the signature, we provide a random number generator,
+	// our private key, the hashing algorithm that we used, and the hash sum
+	// of our message
+	Block, _ := pem.Decode(pemString)
+	privateKey, err := x509.ParsePKCS1PrivateKey(Block.Bytes)
+	if err != nil {
+		return "Failed to parse private key", err
+	}
+
+	signature, err := rsa.SignPKCS1v15(rand.Reader, privateKey, crypto.SHA256, msgHashSum)
+	if err != nil {
+		return "Failed to sign private key", err
+	}
+
+	encodedSignature := base64.StdEncoding.EncodeToString(signature)
+	return fmt.Sprintf("%s.%s", encodedPayload, encodedSignature), nil
+}
+
+func encodeResponse(httpResponse types.HTTPResponse) string {
+	marshaledOnboardingResponse, err := json.Marshal(httpResponse)
```

Comment:
this may not be only used for onboarding, it's generic, just use `res, err` or better create required methods directly on `type.HTTPResponse`